File: dener8/programmers_괄호회전하기.py
'''
괄호의 쌍이 잘 맞아야 함
'''

def solution(s):
    answer = 0

    for i in range(len(s)):
        new = s[i:] + s[:i]
        stack = []
        for alpha in new:
            if not stack:
                stack.append(alpha)
                continue
            if (stack[-1] == '{' and alpha == '}') or (stack[-1] == '(' and alpha == ')') or stack[
                -1] == '[' and alpha == ']':
                stack.pop()
            else:
                stack.append(alpha)
        if not stack:
            answer += 1

    return answer

File: myunghyunnero/programmers_괄호회전하기.cpp
#include <string>
#include <vector>
#include <stack>
using namespace std;

bool checkRight(string s){
    stack<char> stk;
    for(char c : s){
        if(stk.empty()){
            stk.push(c);
        }else{
            char top = stk.top();
            if((c==']' && top == '[')
              || (c==')' && top == '(')
              || (c=='}' && top == '{')){
                stk.pop();
                continue;
            }
            stk.push(c);
        }
    }
    if(stk.empty()){
        return true;
    }
    return false;
}
int solution(string s) {
    int answer = 0;
    int size = s.size();
    for(int i=0;i<size;i++){
        if(checkRight(s)){
            answer++;
        }
        char tmp = s[0];
        for(int j=0;j<size-1;j++){
            s[j]=s[j+1];
        }
        s[size-1]=tmp;
    }
    return answer;
}